src/providers/OrganizationProvider.tsx

```diff
@@ -107,19 +107,17 @@ export default function OrganizationProvider({ children }: OrganizationProviderP
     if (organizations.length && userPreferences) {
       const organizationId = query.get('organizationId');
       const querySelectionOrg = organizationId && organizations.find((org) => org.id === parseInt(organizationId, 10));
-      setSelectedOrganization((previouslySelectedOrg: Organization | undefined) => {
-        let orgToUse = querySelectionOrg || organizations.find((org) => org.id === previouslySelectedOrg?.id);
-        if (!orgToUse && userPreferences.lastVisitedOrg) {
-          orgToUse = organizations.find((org) => org.id === userPreferences.lastVisitedOrg);
-        }
-        if (!orgToUse) {
-          orgToUse = organizations[0];
-        }
-        if (orgToUse && userPreferences?.lastVisitedOrg !== orgToUse.id) {
```

Comment:
@constanzauanini i removed the check here and we always set the preference now, this was a bug - either that or we reload user preferences after every update. either way, this code is getting messy, i will clean it up to use cached preferences later. for now removing this check fixes a bug.. will get this.

Reply:
Ok, sounds good. Does setting the preference always triggers a lot of data fetches also?

Reply:
Not here because we aren't refreshing the preferences.